samza-core/src/main/java/org/apache/samza/execution/JobGraphJsonGenerator.java

```diff
@@ -119,13 +136,18 @@
     JobGraphJson jobGraphJson = new JobGraphJson();
 
     // build StreamEdge JSON
-    jobGraphJson.streams = new HashMap<>();
-    jobGraph.getSources().forEach(e -> getOrCreateStreamEdgeJson(e, jobGraphJson.streams));
-    jobGraph.getSinks().forEach(e -> getOrCreateStreamEdgeJson(e, jobGraphJson.streams));
-    jobGraph.getIntermediateStreamEdges().forEach(e -> getOrCreateStreamEdgeJson(e, jobGraphJson.streams));
+    ApplicationConfig appConfig = jobGraph.getApplicationConfig();
+    jobGraphJson.applicationName = appConfig.getAppName();
```

Comment:
Same comment as for other places where we're mutating DAG nodes. It's usually safer to pre-create these and pass as constructor param to avoid accidentally forgetting to set a field. It's also easier to understand since the object is immutable once created. Would it be difficult to do that here?